apollo-router/src/plugins/traffic_shaping/mod.rs

```diff
@@ -266,15 +267,7 @@ impl Plugin for TrafficShaping {
 pub(crate) type TrafficShapingSubgraphFuture<S> = Either<
     Either<
         BoxFuture<'static, Result<subgraph::Response, BoxError>>,
-        timeout::future::ResponseFuture<
-            Oneshot<
-                Either<
-                    Retry<RetryPolicy, Either<rate::service::RateLimit<S>, S>>,
-                    Either<rate::service::RateLimit<S>, S>,
-                >,
-                subgraph::Request,
-            >,
-        >,
+        BoxFuture<'static, Result<subgraph::Response, BoxError>>,
```

Comment:
If it's 
```
Either<
        BoxFuture<'static, Result<subgraph::Response, BoxError>>,
        BoxFuture<'static, Result<subgraph::Response, BoxError>>,
    >
```

then we probably don't need Either at all

Reply:
Sadly we can't remove it 😢 
We are using `Either` here: 
https://github.com/apollographql/router/blob/bf66e7ab035be9bcc26a01944710a9199bd3d19e/apollo-router/src/plugins/traffic_shaping/mod.rs#L378
And it forces `Either` on future:
https://docs.rs/tower/latest/src/tower/util/either.rs.html#38

Reply:
I'll take a deeper look but maybe we can completely get rid of either at that place as it's the same kind of future

Reply:
@bnjjj It's the same kind of future, but services are different.
My initial revision removed it all and just used `BoxService.` I also wanted to use proper plugin API.
But @Geal provided some context in the Slack thread.
So, I just reverted my changes to the smallest possible change.